loglib/logreader.c

```diff
@@ -129,8 +129,15 @@ static int binlog_convert_event(bytestream * s, void * ptr)
         }
 
         while (ret == 0 && bytestream_remain(s) > 0) {
-            uint64_t len = 0;
-            ret |= byteread_vint(s, &len);
+
+            uint64_t len_read = 0;
+            ret |= byteread_vint(s, &len_read);
+
+            size_t len = (size_t)len_read;
+            if (len != len_read) {
+                ret = -1;
+                break;
+            }
 
```

Comment:
That works. "byteread_vsize(s, &len)" would also have worked.

Reply:
Actually, why not use the function `byteread_vlen(s, &len)` since you created it?

Reply:
Good catch :) I've changed it.